### PR TITLE
Refactor and relocate connect_to_tor

### PR DESCRIPTION
**Pull request by aagbsn**
this unifies some code and makes sure it is exercised by a unit test

### CODE REVIEW

File: test/template.py [hunk 1]
```diff
@@ -7,16 +7,15 @@
...
 
 
 class TorTestCase(unittest.TestCase):
 
     @defer.inlineCallbacks
     def setUp(self):
-        self.tor = yield build_tor_connection(
-            TCP4ClientEndpoint(reactor, 'localhost', int(
-                os.environ.get('CHUTNEY_CONTROL_PORT'))))
+        self.tor = yield connect_to_tor(False, 30, 30,
```
Comment (DonnchaC) [inline test/template.py]: I would specify the key with the keyword arguments to make it clear what each value represents.

Comment (aagbsn) [reply test/template.py]: done

File: bwscanner/attacher.py [hunk 2]
```diff
@@ -162,3 +147,46 @@ def add_single_port_exit(port):
...
+
+    Configure Tor with the passed options and return a Deferred
+    """
+    # Options for spawned or running Tor to load the correct descriptors.
+    tor_options = {
+        'LearnCircuitBuildTimeout': 0,  # Disable adaptive circuit timeouts.
+        'CircuitBuildTimeout': circuit_build_timeout,
+        'CircuitIdleTimeout': circuit_idle_timeout,
+        'UseEntryGuards': 0,  # Disable UseEntryGuards to avoid PathBias warnings.
+        'UseMicroDescriptors': 0,
```
Comment (DonnchaC) [inline bwscanner/attacher.py:182]: Setting UseMicroDescriptors causes Tor to fetch new descriptors and Txtorcon then updates it's TorState object. There is a race where this state update can cause the tests to fail.

**Comment by aagbsn** [comment]
It looks like we fixed a few more bugs than this PR indicates. In particular:
1. on systems with a local system tor and control socket, build_local_tor_connection would pick the system-tor (this is wrapped in a try/except).

2. set_conf was used to set UseMicroDescriptors to 0, which caused a race between bwscanner and tor fetching more descriptors. Now we wait until a NEWCONSENSUS event has been emitted before proceeding.

**Comment by aagbsn** [comment]
Closing this PR in favor of opening a new PR with DonnchaC's refactoring

**Pull request closed without merging**